### PR TITLE
feat: add util/extensions feature

### PR DESCRIPTION
**Pull request by weiraneve**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist
1. [ ] Makes your submission pass tests?
2. [x] Have you lint your code locally before submission?
3. [ ] Have you updated the wiki of your change?

## Reference
- Card: https://trello.com/c/d3lu8awj

### CODE REVIEW

File: core/src/main/java/com/thoughtworks/ark/core/utils/NetworkUtil.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.thoughtworks.ark.core.utils
+
+import android.content.Context
+import android.content.Context.CONNECTIVITY_SERVICE
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+
+object NetworkUtil {
```
Comment (Tony---Zhang) [inline core/src/main/java/com/thoughtworks/ark/core/utils/NetworkUtil.kt:7]: We have a function `isNetworkReachable` in `NetworkReachableInterceptor`

File: core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.thoughtworks.ark.core.utils
+
+import android.os.Build
+
+object DeviceUtils {
+    fun isEmulator(): Boolean {
+        return (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic")) ||
```
Comment (Tony---Zhang) [inline core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt]: Please inline the function
```
fun isEmulator() = (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic")) ||
....
Build.PRODUCT.contains("simulator")
```

File: core/src/main/java/com/thoughtworks/ark/core/utils/NetworkUtil.kt (hunk 1 above)
Comment (weiraneve) [reply core/src/main/java/com/thoughtworks/ark/core/utils/NetworkUtil.kt:7]: get it

File: core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt (hunk 2 above)
Comment (weiraneve) [reply core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt]: get it

Comment (Tony---Zhang) [reply core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt]: 你没有get，你把那边的方法调用这边提供的工具方法啊。那边的代码写法比这边好一些

Comment (weiraneve) [reply core/src/main/java/com/thoughtworks/ark/core/utils/DeviceUtils.kt]: Sry, I don't know what you mean😂 I will communicate in wecom for it.

**Pull request merged**